src/Avalonia.Base/Media/Fonts/UnmanagedFontMemory.cs

```diff
@@ -0,0 +1,407 @@
+        /// its memory region if found. The method performs bounds checks to ensure the requested table is valid and
+        /// safely accessible. If the table is not found or the font data is invalid, the method returns <see
+        /// langword="false"/>.</remarks>
+        /// <param name="tag">The <see cref="OpenTypeTag"/> identifying the table to retrieve. Must not be <see cref="OpenTypeTag.None"/>.</param>
+        /// <param name="table">When this method returns, contains the memory region of the requested table if the operation succeeds;
+        /// otherwise, contains the default value.</param>
+        /// <returns><see langword="true"/> if the table memory was successfully retrieved; otherwise, <see langword="false"/>.</returns>
+        /// <exception cref="ObjectDisposedException">Thrown if the font memory has been disposed.</exception>
+        public bool TryGetTable(OpenTypeTag tag, out ReadOnlyMemory<byte> table)
+        {
+            table = default;
+
+            // Validate tag
+            if (tag == OpenTypeTag.None)
+            {
+                return false;
+            }
+
+            _lock.EnterUpgradeableReadLock();
+
+            try
+            {
+                if (_disposed)
+                {
+                    throw new ObjectDisposedException(nameof(UnmanagedFontMemory));
+                }
+
+                if (_ptr == IntPtr.Zero || _length < 12)
+                {
+                    return false;
+                }
+
+                // Create a span over the unmanaged memory (read-only view)
+                var fontData = Memory.Span;
+
+                // Minimal SFNT header: 4 (sfnt) + 2 (numTables) + 6 (rest) = 12
+                if (fontData.Length < 12)
+                {
+                    return false;
+                }
+
+                // Check cache first
+                if (_tableCache.TryGetValue(tag, out var cached))
+                {
+                    table = cached;
+
+                    return true;
+                }
+
+                // Parse table directory
+                var numTables = BinaryPrimitives.ReadUInt16BigEndian(fontData.Slice(4, 2));
+                var recordsStart = 12;
+                var requiredDirectoryBytes = checked(recordsStart + numTables * 16);
+
+                if (fontData.Length < requiredDirectoryBytes)
+                {
+                    return false;
+                }
+
+                for (int i = 0; i < numTables; i++)
+                {
+                    var entryOffset = recordsStart + i * 16;
+                    var entrySlice = fontData.Slice(entryOffset, 16);
+                    var entryTag = (OpenTypeTag)BinaryPrimitives.ReadUInt32BigEndian(entrySlice.Slice(0, 4));
+
+                    if (entryTag != tag)
+                    {
+                        continue;
+                    }
+
+                    var offset = (int)BinaryPrimitives.ReadUInt32BigEndian(entrySlice.Slice(8, 4));
```

Comment:
`offset` and `length` should not be cast to `int` right now, as any value `> Int32.MaxValue` will throw instead of simply returning `false` right below.

Proposal: read as signed instead (using `ReadInt32BigEndian`) and return false directly if any of the field is `< 0`.